permissions-map/README.md

```diff
@@ -0,0 +1,213 @@
+# Permissions (Map)
+
+**Note:** This is currently a [_beta_ feature](https://github.com/avenga/couper/blob/master/docs/BETA.md).
+
+Please, see the [Permissions example](../permissions/README.md) before reading this.
+
+Suppose you have a calendar API (like Google's) with (among others) three endpoints:
+* `/calendars/`
+* `/calendars/{calendarId}`
+* `/calendars/{calendarId}/events`
+
+The API should be protected by a `jwt` access control (see example [JWT Access Control](../jwt-access-control/README.md))
+
+Additionally, you want to define certain permissions for the endpoints, depending on the used HTTP method:
+
+* calendar
+* calendar.readonly
+* calendar.events
+* calendar.events.readonly
+
+```hcl
+server {
+  api {
+    base_path = "/calendars"
+    access_control = ["Token"]
+
+    endpoint "/" {
+      beta_required_permission = {
+        POST = "calendar"
+        GET = "calendar.readonly"
+      }
+
+      proxy {
+        backend = "api"
+      }
+    }
+
+    endpoint "/{calendarId}" {
+      beta_required_permission = {
+        GET = "calendar.readonly"
+        PATCH = "calendar"
+        PUT = "calendar"
+        DELETE = "calendar"
+      }
+
+      proxy {
+        backend = "api"
+      }
+    }
+
+    endpoint "/{calendarId}/events" {
+      beta_required_permission = {
+        GET = "calendar.events.readonly"
+        POST = "calendar.events"
+      }
+
+      proxy {
+        backend = "api"
+      }
+    }
+  }
+}
+
+definitions {
+  jwt "Token" {
+    signature_algorithm = "RS256"
+    key_file = "pub-key.pem"
+  }
+
+  backend "api" {
+    origin = "http://api:8080"
+  }
+}
+```
+
+If seen as granted permissions (also called privileges), these permissions are in a hierarchical relationship to one another:
+
+The granted permission calendar.events also "contains" calendar.events.readonly, the granted permission calendar.readonly also "contains" calendar.events.readonly, and the granted permission calendar also "contains" both calendar.readonly and calendar.events.
+
+This can be shown in a graph like this:
+```
+               calendar
+                  |
+        +---------+--------+
+        |                  |
+calendar.readonly  calendar.events
+        |                  |
+        +---------+--------+
+                  |
+      calendar.events.readonly
+```
+
+So, e.g. a client granted the calendar.events permission may also request a route with the required permission calendar.events.readonly, and a client granted the calendar permission may request every route.
+
+In Couper, these relationships are configured using the `jwt` block's `beta_permissions_map` attribute:
+
+```hcl
+  jwt "Token" {
+    signature_algorithm = "RS256"
+    key_file = "pub-key.pem"
+    beta_permissions_map = {    # ←
+      "calendar" = ["calendar.readonly", "calendar.events"]
```

Comment:
`"calendar.events.readonly"` is intentionally missing here as the map is called recursively, right? Maybe add a short remark?